src/benchmarks/real-world/ILLink/BasicBenchmark.cs

```diff
@@ -0,0 +1,237 @@
+using System.Linq;
+using System.Reflection;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Engines;
+
+namespace ILLinkBenchmarks;
+
+[BenchmarkCategory("ILLink")]
+[SimpleJob(RunStrategy.Monitoring, launchCount: 1, warmupCount: 3, targetCount: 20)]
+public partial class BasicBenchmark
+{
+    string[] _args;
+    string _publishOutputFolder;
+    string[] _linkerOutArgs => new string[] { "-out", Path.Combine(_publishOutputFolder, "linked") };
+    string[] _extraArgs = new string[] {
+        "--singlewarn",
+        "--trim-mode", "copy",
+        "--action", "copy",
+        "--singlewarn-", "link6",
+        "--nowarn", "1701;1702;IL2121;1701;1702",
+        "--warn", "5",
+        "--warnaserror-", "--warnaserror", ";NU1605",
+        "--feature", "Microsoft.Extensions.DependencyInjection.VerifyOpenGenericServiceTrimmability", "true",
+        "--feature", "System.ComponentModel.TypeConverter.EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization", "false",
+        "--feature", "System.Resources.ResourceManager.AllowCustomResourceTypes", "false",
+        "--feature", "System.Runtime.InteropServices.BuiltInComInterop.IsSupported", "false",
+        "--feature", "System.Runtime.InteropServices.EnableConsumingManagedCodeFromNativeHosting", "false",
+        "--feature", "System.Runtime.InteropServices.EnableCppCLIHostActivation", "false",
+        "--feature", "System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization", "false",
+        "--feature", "System.StartupHookProvider.IsSupported false",
+        "--feature", "System.Threading.Thread.EnableAutoreleasePool", "false",
+        "--feature", "System.Text.Encoding.EnableUnsafeUTF7Encoding", "false",
+        "-b",
+        "--skip-unresolved", "true" };
+
+    [GlobalSetup(Targets = new[] { nameof(LinkHelloWorld) })]
+    public void LinkHelloWorldGlobalSetup()
+    {
+        string projectFilePath = Environment.GetEnvironmentVariable("ILLINK_SAMPLE_PROJECT");
+        _publishOutputFolder = Utilities.PublishSampleProject(projectFilePath);
+        string rootAssembly = Path.Combine(_publishOutputFolder, "HelloWorld.dll");
+
+        var frameworkFiles = Directory.EnumerateFiles(_publishOutputFolder, "*.dll").Where(a =>
+            FrameworkAssemblies.Contains(Path.GetFileName(a))
+        );
+        var frameworkArgs = frameworkFiles.SelectMany(fileName => new string[] { "-reference", fileName });
+        var assemblyArgs = new string[] { "-a", rootAssembly };
+        _args = assemblyArgs.Concat(frameworkArgs).Concat(_extraArgs).Concat(_linkerOutArgs).ToArray();
+
+        return;
+    }
+
+    [GlobalCleanup(Targets = new[] { nameof(LinkHelloWorld) })]
+    public void LinkHelloWorldGlobalCleanup()
+    {
+        Directory.Delete(_publishOutputFolder, recursive: true);
+    }
+
+    [Benchmark]
+    [BenchmarkCategory("ILLink")]
+    public object LinkHelloWorld()
+    {
+        Mono.Linker.WarnVersion x = Mono.Linker.WarnVersion.Latest;
```

Comment:
Could we do the reflection setup in the GlobalSetup, or is there a reason that we need to do this for every operation? In general, you want to have as little code unrelated to the value you are trying to measure in the benchmark method.

Reply:
Good idea, I'll move that into the setup.